jsonparse.ps1

```diff
@@ -0,0 +1,83 @@
+$location=$args[0]
+$repo=$args[1]
+$name=$args[2]
+
+$location
+$repo
+$name
+$repo.Title
+$JSON = (Get-Content $location) -join "`n" | ConvertFrom-Json
+
+$base = "https://github.com/"
+if($repo.Title -eq "github")
+{
+  $base = "https://github.com/"
+}
+#Renaming the key value of file
+$URI = $JSON.runs
+$JSON.runs.files.PSObject.Properties | foreach-object {
+  $URIReplacement = $_.Name
+  $content = $_.Value
+  $JSON.runs.files.PSObject.Properties.Remove($_.Name)
+  $URIReplacement = $base + $URIReplacement.SubString($URIReplacement.IndexOf($name))
+  $JSON.runs.files | add-member -Name $URIReplacement -Value $content -MemberType NoteProperty
+}
+
+$URI = $JSON.runs.files
+for($i = 0; $i -lt $URI.Count; $i++){
+  $JSON.runs.files[$i].uri = $base + $URI[$i].uri.SubString($URI[$i].uri.IndexOf($name))
+}
+
+# Rewriting all urls for the results locations
+$URI = $JSON.runs.results.locations
+for($i = 0;$i -lt $URI.Count; $i++)
+{
+   $JSON.runs.results.locations[$i].resultFile.uri = $base + $URI[$i].resultFile.uri.SubString($URI[$i].resultFile.uri.IndexOf($name))
+}
+
+$URI = $JSON.runs.results.codeFlows.locations.physicalLocation
+for($i = 0;$i -lt $URI.Count; $i++)
+{
+  $JSON.runs.results.codeFlows.locations.physicalLocation[$i].uri = $base + $URI[$i].uri.SubString($URI[$i].uri.IndexOf($name))
+}
```

Comment:
And finally, note that you have repeated the same replacement logic in several places. I suggest you pull that logic out into a function, say `Rebase-Uri`, and call it everywhere, for example:

```
$location.analysisTarget.uri = Rebase-Uri $location.analysisTarget.uri $githubUriBase #... plus whatever other args it needs
```